tests/ops/op_math/test_decompositions.py

```diff
@@ -984,13 +984,14 @@ def test_two_qubit_decomposition_3_cnots(self, U, wires):
     @pytest.mark.parametrize("U", samples_2_cnots)
     def test_two_qubit_decomposition_2_cnots(self, U, wires):
         """Test that a two-qubit matrix using 2 CNOTs isolation is correctly decomposed."""
+        # NOTE: Currently, we defer to the 3-CNOTs function for the 2-CNOTs case.
 
         U = qml.math.convert_to_su4(np.array(U))
 
         assert _compute_num_cnots(U) == 2
 
         obtained_decomposition = two_qubit_decomposition(U, wires=wires)
-        assert len(obtained_decomposition) == 9
+        assert len(obtained_decomposition) == 11  #  9
 
         tape = qml.tape.QuantumScript(obtained_decomposition)
         obtained_matrix = qml.matrix(tape, wire_order=wires)
```

Comment:
Honestly just combine the test matrices used here with the test case for the 3-cnot case, and remove this test function.

Reply:
Shouldn't we just keep these tests around so we can easily modify them back into two-CNOT mode once we have such a function? :)

Reply:
Oh wait the original 2-CNOT case has been removed. Yeah nvm in that case you can leave this as is, although maybe you can slightly explain what this `# 9` comment means.